rllib/models/utils.py

```diff
@@ -95,41 +98,55 @@ def get_activation_fn(name: Optional[str] = None, framework: str = "tf"):
     if callable(name):
         return name
 
+    name_lower = name.lower() if isinstance(name, str) else name
+
     # Infer the correct activation function from the string specifier.
     if framework == "torch":
-        if name in ["linear", None]:
+        if name_lower in ["linear", None]:
             return None
-        if name == "swish":
+
+        _, nn = try_import_torch()
+        # First try getting the correct activation function from nn directly.
+        # Note that torch activation functions are not all lower case.
+        fn = getattr(nn, name, None)
+        if fn is not None:
+            return fn
+
+        if name_lower in ["swish", "silu"]:
             from ray.rllib.utils.torch_utils import Swish
 
             return Swish
-        _, nn = try_import_torch()
-        if name == "relu":
+        elif name_lower == "relu":
             return nn.ReLU
-        elif name == "tanh":
+        elif name_lower == "tanh":
             return nn.Tanh
-        elif name == "elu":
+        elif name_lower == "elu":
             return nn.ELU
     elif framework == "jax":
-        if name in ["linear", None]:
+        if name_lower in ["linear", None]:
             return None
         jax, _ = try_import_jax()
-        if name == "swish":
+        if name_lower in ["swish", "silu"]:
             return jax.nn.swish
-        if name == "relu":
+        if name_lower == "relu":
             return jax.nn.relu
-        elif name == "tanh":
+        elif name_lower == "tanh":
             return jax.nn.hard_tanh
-        elif name == "elu":
+        elif name_lower == "elu":
             return jax.nn.elu
     else:
         assert framework in ["tf", "tf2"], "Unsupported framework `{}`!".format(
             framework
         )
-        if name in ["linear", None]:
+        if name_lower in ["linear", None]:
             return None
+
         tf1, tf, tfv = try_import_tf()
-        fn = getattr(tf.nn, name, None)
+        # Try getting the correct activation function from tf.nn directly.
+        # Note that tf activation functions are all lower case, so this should always
+        # work.
+        fn = getattr(tf.nn, name_lower, None)
```

Comment:
This is more sturdy now, capturing "SiLU" as well as "silu" (at least for tf, for torch, unfortunately, this doesn't work :( ).